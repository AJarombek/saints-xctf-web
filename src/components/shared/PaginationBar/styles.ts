/**
 * JSS styles for the PaginationBar component.
 * @author Andrew Jarombek
 * @since 7/25/2020
 */

import { FontMixins } from '../../../styles/mixins';
import Colors from '../../../styles/colors';

export default {
  paginationBar: {
    ...FontMixins.robotoBold(),
    display: 'flex',
    alignItems: 'center',
    justifyContent: 'center',
    margin: '30px 0'
  },
  page: {
    cursor: 'pointer',
    padding: '0 12px'
  },
  otherPage: {
    color: '#aaa',
    fontSize: '24px',
    transition: 'color 0.3s ease',

    '&:hover': {
      color: Colors.spotPaletteBrown,
      textDecoration: 'underline',
      textDecorationColor: '#aaa'
    }
  },
  currentPage: {
    color: Colors.sxctfRed,
    fontSize: '28px'
  },
  spread: {
    color: '#aaa',
    fontSize: '24px'
  }
};
