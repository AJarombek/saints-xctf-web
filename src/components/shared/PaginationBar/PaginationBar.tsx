/**
 * DashboardPaginationBar component which paginates the log feed.
 * @author Andrew Jarombek
 * @since 7/25/2020
 */

import React from 'react';
import {createUseStyles} from "react-jss";
import styles from "./styles";
import classNames from "classnames";

interface IProps {
    page: number;
    totalPages: number;
    onChangePage: (page: number) => void;
}

const useStyles = createUseStyles(styles);

const PaginationBar: React.FunctionComponent<IProps> = ({ page, totalPages, onChangePage }) => {
    const classes = useStyles();

    return (
        <div id="paginationBar" className={classes.paginationBar}>
            {page > 3 && (
                <div className={classNames(classes.page, classes.otherPage)} onClick={() => onChangePage(1)}>
                    1
                </div>
            )}
            {page > 4 && (
                <div className={classes.spread}>...</div>
            )}
            {page > 2 && (
                <div className={classNames(classes.page, classes.otherPage)} onClick={() => onChangePage(page - 2)}>
                    {page - 2}
                </div>
            )}
            {page > 1 && (
                <div className={classNames(classes.page, classes.otherPage)} onClick={() => onChangePage(page - 1)}>
                    {page - 1}
                </div>
            )}
            <div className={classNames(classes.page, classes.currentPage)}>
                {page}
            </div>
            {totalPages > page && (
                <div className={classNames(classes.page, classes.otherPage)}  onClick={() => onChangePage(page + 1)}>
                    {page + 1}
                </div>
            )}
            {totalPages > (page + 1) && (
                <div className={classNames(classes.page, classes.otherPage)}  onClick={() => onChangePage(page + 2)}>
                    {page + 2}
                </div>
            )}
        </div>
    );
};

export default PaginationBar;
