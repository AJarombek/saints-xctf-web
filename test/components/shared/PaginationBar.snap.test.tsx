/**
 * Snapshot test for the PaginationBar component.
 * @author Andrew Jarombek
 * @since 3/7/2021
 */

import React from 'react';
import renderer from 'react-test-renderer';
import PaginationBar from '../../../src/components/shared/PaginationBar';

describe('PaginationBar Snapshot Tests', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<PaginationBar page={5} totalPages={10} onChangePage={(): void => null} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
