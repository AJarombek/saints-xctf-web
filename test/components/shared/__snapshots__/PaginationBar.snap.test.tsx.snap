// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PaginationBar Snapshot Tests renders correctly 1`] = `
<div
  className="paginationBar-0-1-1"
  id="paginationBar"
>
  <div
    className="page-0-1-2 otherPage-0-1-3"
    onClick={[Function]}
  >
    1
  </div>
  <div
    className="spread-0-1-5"
  >
    ...
  </div>
  <div
    className="page-0-1-2 otherPage-0-1-3"
    onClick={[Function]}
  >
    3
  </div>
  <div
    className="page-0-1-2 otherPage-0-1-3"
    onClick={[Function]}
  >
    4
  </div>
  <div
    className="page-0-1-2 currentPage-0-1-4"
  >
    5
  </div>
  <div
    className="page-0-1-2 otherPage-0-1-3"
    onClick={[Function]}
  >
    6
  </div>
  <div
    className="page-0-1-2 otherPage-0-1-3"
    onClick={[Function]}
  >
    7
  </div>
</div>
`;
